### DISCUSSION TITLE
Do you have any PMD  rules which allows to forbid the public constants in my project?

### DISCUSSION THREAD

**Discussion by dgroup** [pmd/pmd#3054] [Q&A] [answered]
Hi Team,

Do you have any rules which allows to forbid the public constants in my project?

Thanks a lot,
Yurii

**Comment by jsotuyod** [comment, marked as answer]
There are no such rules out of the box, but thanks to PMD's capabilities to [write rules as XPath expressions](https://pmd.github.io/pmd-5.8.0/customizing/xpathruletutorial.html), you can easily add one yourself [to your ruleset](https://pmd.github.io/pmd-5.8.0/customizing/howtomakearuleset.html):

For instance, an XPath expression such as:

```
//FieldDeclaration[@Public = 'true'][@Static = 'true'][@Final = 'true']
```

would match all fields that are public static final.

**Comment by jsotuyod** [comment]
@dgroup could you get this working? Do you need any further assistance?
